service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementMapper.java

```diff
@@ -41,182 +44,265 @@
+                return referralRequest.getNote().stream()
+                    .map(value -> String.format(NOTE, extractAuthor(messageContext, value), extractNoteTime(value), value.getText()))
+                    .collect(Collectors.joining(COMMA));
+            }
+            return StringUtils.EMPTY;
+        }
 
-    private List<Reference> getRecipientsWithoutFirstPractitioner(ReferralRequest referralRequest) {
-        boolean firstPractitionerFound = false;
-        List<Reference> recipients = new ArrayList<>(referralRequest.getRecipient().size());
-        for (Reference reference : referralRequest.getRecipient()) {
-            if (!firstPractitionerFound && isReferenceToPractitioner(reference)) {
-                firstPractitionerFound = true;
-            } else {
-                recipients.add(reference);
+        private String buildDeviceDescription(Reference agent) {
+            if (isReferenceToType(agent, ResourceType.Device)) {
+                return messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Device.class::isInstance)
+                    .map(Device.class::cast)
+                    .filter(Device::hasType)
+                    .map(Device::getType)
+                    .flatMap(CodeableConceptMappingUtils::extractTextOrCoding)
+                    .map(REQUESTER_DEVICE::concat)
+                    .orElseThrow(() -> new EhrMapperException("Could not resolve Device Reference"));
             }
+            return StringUtils.EMPTY;
         }
-        return recipients;
-    }
 
-    private boolean isReferenceToPractitioner(Reference reference) {
-        return reference.getReference().startsWith(ResourceType.Practitioner.name());
-    }
+        private String buildOrganizationDescription(Reference agent) {
+            if (isReferenceToType(agent, ResourceType.Organization)) {
+                return messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Organization.class::isInstance)
+                    .map(Organization.class::cast)
+                    .filter(Organization::hasName)
+                    .map(Organization::getName)
+                    .map(REQUESTER_ORG::concat)
+                    .orElseThrow(() -> new EhrMapperException("Could not resolve Organization Reference"));
+            }
+            return StringUtils.EMPTY;
+        }
 
-    private String buildReasonCodeDescription(ReferralRequest referralRequest) {
-        if (referralRequest.hasReasonCode() && (referralRequest.getReasonCode().size() > 1)) {
-            return REASON_CODE + extractReasonCode(referralRequest);
+        private String buildPatientDescription(Reference agent) {
+            if (isReferenceToType(agent, ResourceType.Patient)) {
+                return messageContext.getInputBundleHolder().getResource(agent.getReferenceElement())
+                    .filter(Patient.class::isInstance)
```

Comment:
I don't think it is necessary to check if the Patient resource is in the bundle. Checking the reference type seems to be enough, because we do not need any information from Patient resource to extend text.

Reply:
Sure, I'll remove all the filters doing that